inginious/frontend/user_manager.py

```diff
@@ -451,6 +499,54 @@ def bind_user(self, auth_id, user):
 
         return True
 
+    def revoke_binding(self, username, binding_id):
+        """
+        Revoke a binding method for a user
+        :param binding_id: The binding method id
+        :param username: username of the user
+        :return: Boolean if error occurred and message if necessary
+        """
+        user_data = self._database.users.find_one({"username": username})
+        if binding_id not in self.get_auth_methods().keys():
+            error = True
+            msg = _("Incorrect authentication binding.")
+        elif len(user_data.get("bindings", {}).keys()) > 1 or "password" in user_data:
```

Comment:
This will throw an exception if ``username`` is not found in database and ``user_data`` is ``None``.